docs(lapis): document string search feature (#906)
GenSpectrum/LAPIS: lapis-docs/src/content/docs/concepts/string-search.mdx

Reviewer: Maybe we can add something like "by default, LAPIS only allows filtering for exact values. But if the maintainer enables regex search..."?

Reviewer: I think a couple of examples would be good. At least for the most basic example for how to filter for "string contains": `field$regex=substring`.

---
title: String search
description: String search
---

LAPIS allows users to filter string fields in the metadata using [regular expressions](https://en.wikipedia.org/wiki/Regular_expression).
This is useful when the full string is unknown or when a more complex search is needed.

This filter is optional for all string fields and must be enabled by the instance administrator.
A list of searchable fields of this LAPIS instance can be found [here](../references/filters).

The syntax for the string search is a subset of the syntax of the [PCRE](https://www.pcre.org/) library.

> A few examples can be found behind this [link](https://github.com/google/re2/wiki/Syntax).

Reviewer: I'd prefer to be a bit more explicit:
```suggestion
Under the hood, we use the [RE2](https://github.com/google/re2) library to evaluate the regular expressions.
The exact syntax is documented here: [Syntax wiki](https://github.com/google/re2/wiki/Syntax).
```

---
title: String search
description: String search
---

LAPIS allows users to filter string fields in the metadata using [regular expressions](https://en.wikipedia.org/wiki/Regular_expression).
This is useful when the full string is unknown or when a more complex search is needed.

By default, the string search is disabled for all string fields. It can be enabled by the instance administrator.
A list of searchable fields of this LAPIS instance can be found [here](../references/filters).

The syntax for the string search is a subset of the syntax of the [PCRE](https://www.pcre.org/) library.
Under the hood, we use the [RE2](https://github.com/google/re2) library to evaluate the regular expressions.

## Examples

Get the number of sequences with the `division` field containing the string `Basel`:

```

> [URL to LAPIS instance]/sample/aggregated?division$regex=Basel

Reviewer: ```suggestion
GET /sample/aggregated?division$regex=Basel
```

We also used that format in other examples already.

```

Get the number of sequences with the `division` field starting with the string `Basel`:

```

> [URL to LAPIS instance]/sample/aggregated?division$regex=^Basel

Reviewer: ```suggestion
GET /sample/aggregated?division$regex=^Basel
```
